### PR TITLE
Data loss warning on Angular editforms on window.beforeunload

### PR DESCRIPTION
**Pull request by aleixsuau**
https://community.openproject.com/projects/openproject/work_packages/25683

This pull request:

- [x] Warns the user about a possible data loss on the Angular forms (EditFormComponent) when moving out of the  Angular app (ie: clicking on the main side menu items).
- [x] Adds a GlobalEditFormChangesTrackerService that tracks the changes on all the EditFormComponent active in the app.

Note:
This pull request solves the bug but only on the changes made inside and EditFormComponent (ie: wp description). Changes made out of them, like comments on a work package, are not registered on the GlobalEditFormChangesTrackerService and would be lost without any warning when clicking on a link out of the Angular app.

I created a [new work package](https://community.openproject.com/projects/openproject/work_packages/34551) to cover these cases too.

### CODE REVIEW

File: frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts [hunk 1]
```diff
@@ -44,6 +44,7 @@ import {EditFieldHandler} from "core-app/modules/fields/edit/editing-portal/edit
 import {EditingPortalService} from "core-app/modules/fields/edit/editing-portal/editing-portal-service";
 import {EditFormRoutingService} from "core-app/modules/fields/edit/edit-form/edit-form-routing.service";
 import {ResourceChangesetCommit} from "core-app/modules/fields/edit/services/hal-resource-editing.service";
+import {GlobalEditFormChangesTrackerService} from "core-app/modules/fields/edit/services/global-edit-form-changes-tracker/global-edit-form-changes-tracker.service";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts:48]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/7a895a2f422ccb1590b4607212148b670ce61d59/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts [hunk 2]
```diff
@@ -44,6 +44,8 @@ import {EditFieldHandler} from "core-app/modules/fields/edit/editing-portal/edit
...
+import {FormattableEditFieldComponent} from "core-app/modules/fields/edit/field-types/formattable-edit-field.component";
```
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts:49]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/531e5548faf70fd912e127df04fbdb69ca54c754/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L49"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

File: frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts (hunk 1 above)
Comment (openprojectci) [inline frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts:48]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/83f0f3ca335b769d053227115d7648f66a062b1d/frontend/src/app/modules/fields/edit/edit-form/edit-form.component.ts#L48"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>

**Comment by aleixsuau** [comment]
Closing because creating a new branch rebased on release/11.0

**Pull request closed without merging**